Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
```

Comment:
Not your fault, but it feels like `PythonRequirement` and `Python3Requirement` having a parent-child relationship instead of being siblings is a design flaw that is exposed by this.